File: Leetcode/94. Binary Tree Inorder Traversal.py
class Solution:
    def __init__(self):
        self.ans = []
    def f(self,root):
        if not root:
            return
        self.f(root.left)
        self.ans.append(root.val)
        self.f(root.right)
    def inorderTraversal(self, root: Optional[TreeNode]) -> List[int]:
        self.f(root)
        return self.ans
        

        

File: LC/94. Binary Tree Inorder Traversal.cpp
class Solution {
private:
    void f(TreeNode* root,vector<int> &ans) {
        if(!root) return;
        f(root->left,ans);
        ans.push_back(root->val);
        f(root->right,ans);
    }
public:
    vector<int> inorderTraversal(TreeNode* root) {
        vector<int> ans;
        f(root,ans);
        return ans;
    }
};